<?xml version="1.0"?>
<launch>
  <arg name="rigid_only" default="true" doc="set to false to allow soft bodies"/>
  <arg name="use_rviz" default="true"/>
  <arg name="use_sim_clock" default="false"/>
  <arg name="step_time" default="0.03" doc="how fast the simulation is updating externally"/>
  <arg name="tf_prefix" default=""/>
  <arg name="gravity_z" default="-9.8" doc="gravity in units/sec^2"/>

  <param name="use_sim_time" value="$(arg use_sim_clock)"/>
  <group if="$(arg use_sim_clock)">
    <node name="sim_clock" pkg="sim_clock" type="clock">
      <param name="dt" value="0.01"/>
      <param name="step_time" value="$(arg step_time)"/>
    </node>
    <node name="rqt_gui" pkg="rqt_gui" type="rqt_gui"/>
  </group>

  <node pkg="rviz" type="rviz" name="rviz"
      args="-d $(find bullet_server)/config/bullet_server.rviz"
      if="$(arg use_rviz)"/>

  <node pkg="bullet_server" type="bullet_server" name="bullet_server"
      output="screen">
    <param name="rigid_only" value="$(arg rigid_only)"/>
    <param name="tf_prefix" value="$(arg tf_prefix)"/>
    <param name="gravity" value="$(arg gravity)"/>
    <param name="target_time_step" value="$(arg step_time)"/>
    <param name="fixed_time_step" value="0.0003"/>
    <param name="max_sub_steps" value="150"/>
  </node>

</launch>
